enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,597 @@
+controller, this is responsible for updating the cluster `ETCD` resource with
+the correct `observedConfig.controlPlane.replicas`, this needs to be updated to
+correctly reflect the `replicas` count that contains both `controlPlane` and
+`arbiterNode` counts.
+
+We have a few more changes in CEO related to the deligated tooling in
+`library-go`, that will be covered in the next section.
+
+### library-go Change
+
+Continuing from the CEO change, some of the underlying controllers used by CEO
+need small changes to support the arbiter node. The `StaticPod` controller and
+the `Guard` controller will need to be updated to add a helper method that
+allows CEO and other tools to easily turn on the `arbiter` node selector for
+their internal node listers.
+
+`ex:`
+
+```go
+
+func (c *Controller) WithArbiterNode() {
+    c.withArbiterNode = true;
+}
+...
+
+if c.withArbiterNode {
+    arbiterselector, err := labels.NewRequirement("node-role.kubernetes.io/arbiter", selection.Equals, []string{""})
+    arbiternodes, err := c.nodeLister.List(labels.NewSelector().Add(*arbiterselector))
+    nodes = append(nodes, arbiternodes...)
+}
+...
+
+```
+
+### Authentication Operator Change
+
+The authentication operator runs a [readiness check
+controller](https://github.com/openshift/cluster-authentication-operator/blob/2d71f164af3f3e9c84eb40669d330df2871956f5/pkg/controllers/readiness/unsupported_override.go#L58-L70)
+that enforces for all non `SingleReplicaTopologyMode` topologies or an
+explicitly override, there must be 3 or more `kube-apiservers`. We need to add
+another check to this switch statement to allow minimum 2 replicas for
+`HighlyAvailableArbiter` topologies.
+
+### Workflow Description
+
+#### For Cloud Installs
+
+1. The user creates an `install-config.yaml`.
+2. The user defines the `installConfig.controlPlane` field with `2` replicas.
+3. The user then enters the new field `installConfig.arbiterNode` defines the
+   arbiter node.
+4. The user generates the manifests with this install config via the
+   `openshift-install create manifests`
+5. The installer creates a new `arbiter` MachineSet with a replica of `1` and
+   checks the control plane replicas is `2` or higher
+6. The installer sets the control plane and infrastructure topology to
+   `HighlyAvailableArbiter`
+7. With the object `arbiterNode` in the install config, the installer creates
+   the machine object with the correct node labels and machine config resources
+   for ignition bootstrap files.
+8. The installer applies the taint to the arbiter MachineSet
+9. The user can make any alterations to the node machine type to use less
+   powerful machines.
+10. The user then begins the install via `openshift-install create cluster`
+
+#### For Baremetal Installs
```

Comment:
We should also be thinking about how to add support for this in the assisted installer, ZTP, and the agent-based installer.
One of our goals at the moment is to close all of the gaps between metal IPI and ABI, with a view to incorporating the latter into the former. So new gaps will not be welcomed :slightly_smiling_face:

Reply:
Fully agree, I'll add a section on that path as well, I don't have a clear idea on that due to unfamiliarity with assisted. I'll document our desire here and update early next week after some research.